map.go

```diff
@@ -1622,12 +1649,56 @@ func newMapIterator(target *Map) *MapIterator {
 	}
 }
 
+// cursorToKeyOut copies the current value held in the cursor to the
+// provided argument. In case of errors, returns false and sets a
+// non-nil error in the MapIterator.
+func (mi *MapIterator) cursorToKeyOut(keyOut interface{}) bool {
+	buf := mi.cursor.([]byte)
+	if ptr, ok := keyOut.(unsafe.Pointer); ok {
+		copy(unsafe.Slice((*byte)(ptr), len(buf)), buf)
+	} else {
```

Comment:
Sorry if this is a bit off topic. @ti-mo Do you recall why we have this exception here? This is just moving the existing code added in https://github.com/cilium/ebpf/commit/4609dc73fa846184b379ae1ee771cc9e95e2ecbc. But we seem to take so much care to ensure safety elsewhere, including in `sysenc.Unmarshal`. But not here.

If for some reason the user passes in an unsafe pointer in `keyOut` that is smaller than `buf` this causes us to write into unknown parts of the heap.